imxrt-hal/src/tempmon.rs

```diff
@@ -0,0 +1,395 @@
+#![cfg(any(feature = "imxrt1061", feature = "imxrt1062", feature = "imxrt1064"))]
+
+//! # Temperature Monitor (TEMPMON)
+//!
+//! ## IMPORTANT NOTE:
+//! The temperature sensor uses and assumes that the bandgap
+//! reference, 480MHz PLL and 32KHz RTC modules are properly programmed and fully
+//! settled for correct operation.
+//!
+//! ## Example 1
+//!
+//! Manually triggered read
+//!
+//! ```no_run
+//! use imxrt_hal;
+//!
+//! let mut peripherals = imxrt_hal::Peripherals::take().unwrap();
+//!
+//! let (_, ipg_hz) = peripherals.ccm.pll1.set_arm_clock(
+//!     imxrt_hal::ccm::PLL1::ARM_HZ,
+//!     &mut peripherals.ccm.handle,
+//!     &mut peripherals.dcdc,
+//! );
+//!
+//! let mut cfg = peripherals.ccm.perclk.configure(
+//!     &mut peripherals.ccm.handle,
+//!     imxrt_hal::ccm::perclk::PODF::DIVIDE_3,
+//!     imxrt_hal::ccm::perclk::CLKSEL::IPG(ipg_hz),
+//! );
+//!
+//! // init temperature monitor
+//! let mut temp_mon = peripherals.tempmon.init();
+//! loop {
+//!     if let Ok(temperature) = nb::block!(temp_mon.measure_temp()) {
+//!         // temperature in mC (1°C = 1000°mC)
+//!     }
+//! }
+//! ```
+//!
+//! ## Example 2
+//!
+//! Non-blocking reading
+//!
+//! ```ignore
```

Comment:
Could we make this a `no_run` example, and include the necessary code to make it compile? We could [hide portions of the example 1 code](https://doc.rust-lang.org/rustdoc/documentation-tests.html#hiding-portions-of-the-example) to keep the example focused on the new content.

Question also applies to example 3, below.